File: app/src/main/java/com/six/cat/sixcat/utils/SettingUtil.java
package com.six.cat.sixcat.utils;

import android.graphics.Color;

import com.six.cat.sixcat.R;
import com.six.cat.sixcat.SixCatApplication;

/**
 * @author liguoying
 * @date 2017/12/11.
 */

public class SettingUtil {

    private static final class SettingsUtilInstance {
        private static final SettingUtil instance = new SettingUtil();
    }

    public static SettingUtil getInstance() {
        return SettingsUtilInstance.instance;
    }

    /**
     * 获取是否开启无图模式
     */
    public boolean getIsNoPhotoMode() {
        return SPUtil.getBoolean("switch_noPhotoMode", false) && NetWorkUtil.isMobileConnected(SixCatApplication.getInstance());
    }

    /**
     * 获取主题颜色
     */
    public int getColor() {
        int defaultColor = SixCatApplication.getInstance().getResources().getColor(R.color.colorPrimary, null);
        int color = SPUtil.getInt("color", defaultColor);
        if ((color != 0) && Color.alpha(color) != 255) {
            return defaultColor;
        }
        return color;
    }

    /**
     * 设置主题颜色
     */
    public void setColor(int color) {
        SPUtil.setInt("color", color);
    }

    /**
     * 获取是否开启夜间模式
     */
    public boolean getIsNightMode() {
        return SPUtil.getBoolean("switch_nightMode", false);
    }

    /**
     * 设置夜间模式
     */
    public void setIsNightMode(boolean flag) {
        SPUtil.setBoolean("switch_nightMode", flag);
    }

    /**
     * 获取是否开启自动切换夜间模式
     */
    public boolean getIsAutoNightMode() {
        return SPUtil.getBoolean("auto_nightMode", false);
    }

    /**
     * 获取是否开启自动切换夜间模式
     */
    public void setIsAutoNightMode(boolean flag) {
        SPUtil.setBoolean("auto_nightMode", flag);
    }

    public String getNightStartHour() {
        return SPUtil.getString("night_startHour", "22");
    }

    public void setNightStartHour(String nightStartHour) {
        SPUtil.setString("night_startHour", nightStartHour);
    }

    public String getNightStartMinute() {
        return SPUtil.getString("night_startMinute", "00");
    }

    public void setNightStartMinute(String nightStartMinute) {
        SPUtil.setString("night_startMinute", nightStartMinute);
    }

    public String getDayStartHour() {
        return SPUtil.getString("day_startHour", "06");
    }

    public void setDayStartHour(String dayStartHour) {
        SPUtil.setString("day_startHour", dayStartHour);
    }

    public String getDayStartMinute() {
        return SPUtil.getString("day_startMinute", "00");
    }

    public void setDayStartMinute(String dayStartMinute) {
        SPUtil.setString("day_startMinute", dayStartMinute);
    }

    /**
     * 获取是否开启导航栏上色
     */
    public boolean getNavBar() {
        return SPUtil.getBoolean("nav_bar", false);
    }

    /**
     * 获取是否开启视频强制横屏
     */
    public boolean getIsVideoForceLandscape() {
        return SPUtil.getBoolean("video_force_landscape", false);
    }

    /**
     * 获取图标值
     */
    public int getCustomIconValue() {
        String s = SPUtil.getString("custom_icon", "0");
        return Integer.parseInt(s);
    }

    /**
     * 获取滑动返回值
     */
    public int getSlidable() {
        String s = SPUtil.getString("slidable", "1");
        return Integer.parseInt(s);
    }

    /**
     * 获取是否开启视频自动播放
     */
    public boolean getIsVideoAutoPlay() {
        return SPUtil.getBoolean("video_auto_play", false) && NetWorkUtil.isWifiConnected(SixCatApplication.getInstance());
    }

    /**
     * 获取字体大小
     */
    public int getTextSize() {
        return SPUtil.getInt("textsize", 16);
    }

    /**
     * 设置字体大小
     */
    public void setTextSize(int textSize) {
        SPUtil.setInt("textsize", textSize);
    }

    public boolean getIsFirstTime() {
        return SPUtil.getBoolean("first_time", true);
    }

    public void setIsFirstTime(boolean flag) {
        SPUtil.setBoolean("first_time", flag);
    }

}

File: app/src/main/java/com/sixcat/utils/SettingUtil.kt
package com.sixcat.utils

import android.graphics.Color

import com.sixcat.R
import com.sixcat.SixCatApplication

/**
 * @author liguoying
 * @date 2017/12/11.
 */

class SettingUtil {

    /**
     * 获取是否开启无图模式
     */
    val isNoPhotoMode: Boolean
        get() = SPUtil.getBoolean("switch_noPhotoMode", false) && NetWorkUtil.isMobileConnected(SixCatApplication.context)

    /**
     * 获取主题颜色
     */
    /**
     * 设置主题颜色
     */
    var color: Int
        get() {
            val defaultColor = SixCatApplication.context.resources.getColor(R.color.colorPrimary, null)
            val color = SPUtil.getInt("color", defaultColor)
            return if (color != 0 && Color.alpha(color) != 255) {
                defaultColor
            } else color
        }
        set(color) = SPUtil.setInt("color", color)

    /**
     * 获取是否开启夜间模式
     */
    /**
     * 设置夜间模式
     */
    var isNightMode: Boolean
        get() = SPUtil.getBoolean("switch_nightMode", false)
        set(flag) = SPUtil.setBoolean("switch_nightMode", flag)

    /**
     * 获取是否开启自动切换夜间模式
     */
    /**
     * 获取是否开启自动切换夜间模式
     */
    var isAutoNightMode: Boolean
        get() = SPUtil.getBoolean("auto_nightMode", false)
        set(flag) = SPUtil.setBoolean("auto_nightMode", flag)

    var nightStartHour: String
        get() = SPUtil.getString("night_startHour", "22")
        set(nightStartHour) = SPUtil.setString("night_startHour", nightStartHour)

    var nightStartMinute: String
        get() = SPUtil.getString("night_startMinute", "00")
        set(nightStartMinute) = SPUtil.setString("night_startMinute", nightStartMinute)

    var dayStartHour: String
        get() = SPUtil.getString("day_startHour", "06")
        set(dayStartHour) = SPUtil.setString("day_startHour", dayStartHour)

    var dayStartMinute: String
        get() = SPUtil.getString("day_startMinute", "00")
        set(dayStartMinute) = SPUtil.setString("day_startMinute", dayStartMinute)

    /**
     * 获取是否开启导航栏上色
     */
    val navBar: Boolean
        get() = SPUtil.getBoolean("nav_bar", false)

    /**
     * 获取是否开启视频强制横屏
     */
    val isVideoForceLandscape: Boolean
        get() = SPUtil.getBoolean("video_force_landscape", false)

    /**
     * 获取图标值
     */
    val customIconValue: Int
        get() {
            val s = SPUtil.getString("custom_icon", "0")
            return Integer.parseInt(s)
        }

    /**
     * 获取滑动返回值
     */
    val slidable: Int
        get() {
            val s = SPUtil.getString("slidable", "1")
            return Integer.parseInt(s)
        }

    /**
     * 获取是否开启视频自动播放
     */
    val isVideoAutoPlay: Boolean
        get() = SPUtil.getBoolean("video_auto_play", false) && NetWorkUtil.isWifiConnected(SixCatApplication.context)

    /**
     * 获取字体大小
     */
    /**
     * 设置字体大小
     */
    var textSize: Int
        get() = SPUtil.getInt("textsize", 16)
        set(textSize) = SPUtil.setInt("textsize", textSize)

    var isFirstTime: Boolean
        get() = SPUtil.getBoolean("first_time", true)
        set(flag) = SPUtil.setBoolean("first_time", flag)

    private object SettingsUtilInstance {
        val instance = SettingUtil()
    }

    companion object {
        val instance: SettingUtil
            get() = SettingsUtilInstance.instance
    }

}

